### PR TITLE
Fixed apache-geode forumla to not use Java 1.9

### PR DESCRIPTION
**Pull request by jaredjstewart**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

 I encountered an error when running `brew audit --strict foo`:

```
apache-geode:
  * version should not decrease
Error: 1 problem in 1 formula
```
-----

### CODE REVIEW

**Comment by MikeMcQuaid** [comment]
Can you elaborate a bit more here on why we need to avoid Java 1.9?

**Comment by jaredjstewart** [comment]
`gfsh start locator` does not return to foreground, just lots of `…………………………….`

**Comment by MikeMcQuaid** [comment]
@jaredjstewart Please file an upstream issue about this, thanks.

**Comment by jaredjstewart** [comment]
@MikeMcQuaid Do you mean upstream with Apache Geode?

File: Formula/apache-geode.rb [hunk 1]
```diff
@@ -7,13 +7,13 @@ class ApacheGeode < Formula
 
   bottle :unneeded
 
-  depends_on :java => "1.8+"
+  depends_on :java => "1.8"
```
Comment (MikeMcQuaid) [inline Formula/apache-geode.rb]: Please link to `https://issues.apache.org/jira/browse/GEODE-3` and explain that it doesn't work with Java 1.9 on the line above. Thanks!

**Review by MikeMcQuaid — CHANGES_REQUESTED** [review]

File: Formula/apache-geode.rb (hunk 1 above)
Comment (jaredjstewart) [inline Formula/apache-geode.rb]: I just pushed another commit to include this comment.

File: Formula/apache-geode.rb [hunk 2]
```diff
@@ -7,13 +7,14 @@ class ApacheGeode < Formula
 
   bottle :unneeded
 
-  depends_on :java => "1.8+"
+  #Geode does not work with Java 1.9 (see https://issues.apache.org/jira/browse/GEODE-3).
```
Comment (MikeMcQuaid) [inline Formula/apache-geode.rb]: `brew style` warning: https://bot.brew.sh/job/Homebrew%20Core/11526/version=any_sierra/testReport/junit/brew-test-bot/sierra/style_apache_geode/

**Review by MikeMcQuaid — CHANGES_REQUESTED** [review]

File: Formula/apache-geode.rb (hunk 2 above)
Comment (MikeMcQuaid) [inline Formula/apache-geode.rb]: Also you can skip the trailing `.`

**Comment by jaredjstewart** [comment]
I updated the PR, comment style should now be compliant:
```
$ brew style apache-geode

1 file inspected, no offenses detected
```

**Comment by dunn** [comment]
Thanks, merged as a3f1f3d9af9ca1ff0d8015375bdc575e5a35c2a5!

**Pull request closed without merging**